pkg/crds/controllers/batch/batchjob_controller_helpers.go

```diff
@@ -672,10 +685,11 @@ func getMetrics(r *BatchJobReconciler, batchJob batch.BatchJob) (metrics.BatchMe
 		ID:      batchJob.Name,
 		APIName: batchJob.Spec.APIName,
 		Kind:    userconfig.BatchAPIKind,
-	}, batchJob.Status.EndTime.Time)
+	}, time.Now())
```

Comment:
shouldn't this be something like:

```go
endTime := time.Now
if batchJob.Status.EndTime != nil {
    endTime = batchJob.Status.EndTime.Time
}
```
and then we use the `endTime` variable in the call to `GetMetrics`

Reply:
Initially, I was thinking `time.Now()` is generally a good option. Semantically, using endTime makes sense. Agreed.